src/parse-stream.js

```diff
@@ -427,4 +434,39 @@ export default class ParseStream extends Stream {
       data: line.slice(4)
     });
   }
+
+  /**
+   * Add a parser for custom headers
+   *
+   * @param {RegExp}   expression  a regular expression to match the custom header
+   * @param {string}   customType  the custom type to register to the output
+   * @param {Function} dataParser  function to parse the line into an object
+   * @param {boolean}  segment     should the tag be put into the segment data
+   */
+  addParser(expression, customType, dataParser, segment) {
+    if (typeof dataParser !== 'function') {
+      if (typeof dataParser === 'boolean') {
```

Comment:
I think it might be clearer to either have someone setting up a segment tag parser always have to have the dataParser argument, or we should change the addParser method to accept an object instead, so this can just be `addParser({expression, customType, dataParser, isSegmentTag})` or something like that